packages/components/src/hooks/usePasswordValidation.tsx

```diff
@@ -0,0 +1,121 @@
+import { Box, Flex, Text } from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { z } from "zod";
+import zxcvbn from "zxcvbn";
+
+const DEFAULT_SCORE = 0;
+const DEFAULT_COLOR = "gray.100";
+const DEFAULT_MIN_LENGTH = 12;
+const DEFAULT_PASSWORD_FIELD_NAME = "password";
+
+type PasswordStrengthBarProps = {
+  score: number;
+  color: string;
+  hasError: boolean;
+};
+
+type UsePasswordValidationProps = {
+  color?: string;
```

Comment:
maybe defaultBarColor or something similar? Not very obvious what this param is for with the current name

Reply:
I think `color` would be more concise in this case, given that the only place where color can be used as a prop in this hook is the PasswordStrengthBar, so I don't think it can be too confusing